### PR TITLE
Core: Replace global Qt declarations with standard equivalents

### PR DESCRIPTION
**Pull request by sakertooth**
Replaces the functions ``qBound``, ``qMax``, ``qMin``, ``qRound``, and ``qgetenv`` with ``std::clamp``, ``std::max``, ``std::min``, ``std::round``, and ``std::getenv`` respectively. These were the only functions used within the core that were provided by Qt's global declarations.

### CODE REVIEW

**Review by DomClark — APPROVED** [review]
There's a slight change in behaviour here with the environment variable handling: previously, empty environment variables were ignored as if they were not set, but now the empty value is used. I'm not sure whether this really matters, but thought it was worth mentioning. Otherwise, LGTM.

File: src/core/Note.cpp [hunk 1]
```diff
@@ -114,7 +114,7 @@ void Note::setPos( const TimePos & pos )
 
 void Note::setKey( const int key )
 {
-	const int k = qBound( 0, key, NumKeys - 1 );
```
Comment (DomClark) [inline src/core/Note.cpp:117]: This upper bound is different to the one in the constructor, which uses `NumKeys`. Out of scope here, of course, but probably should be investigated.

File: src/core/PluginFactory.cpp [hunk 2]
```diff
@@ -86,7 +86,7 @@ void PluginFactory::setupSearchPaths()
 #endif
 	// Or via an environment variable:
 	QString env_path;
```
Comment (messmerd) [inline src/core/PluginFactory.cpp:88]:
```suggestion
```
Looks like this QString can be removed.

Comment (sakertooth) [reply src/core/PluginFactory.cpp:88]: Oh, I should've assigned it to that variable. If its unused I'll try to remove it soon. Thanks for the heads up.

**Review by messmerd — APPROVED** [review]

**Pull request merged**